### DISCUSSION TITLE
Fontawesome dynamic import

### DISCUSSION THREAD

**Discussion by Sandeep06Dev** [vercel/next.js#16027] [Help]
How to import React font awesome all icons back with lazy / dynamic load. As react fontawesome pack is around 600kb huge..

So i want to use it dynamic / lazy load.

**Comment by marcmll** [comment]
Do you need all of the icons?
Fontawesome gives you the option to add icons one by one, dramatically decreasing what is being loaded.

For example, if you are looking to only import social icons, your `_app.js` could look something like this:
```js
// Import FontAwesome Icons
import { library } from '@fortawesome/fontawesome-svg-core'
import { faGithub, faLinkedin, faTwitter } from '@fortawesome/free-brands-svg-icons'
library.add(faGithub, faLinkedin, faTwitter)

export default function MyApp ({ Component, pageProps }) {
  return <Component {...pageProps} />
}
```

The icons can then be used the same way they would be if you imported the entire pack:
```js
// pages/index.js
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

export default function HomePage () {
  return (
    <FontAwesomeIcon icon={['fab', 'twitter']} />
  )
}
```

More info: https://fontawesome.com/how-to-use/on-the-web/using-with/react#using

**Comment by drskullster** [comment]
If you have a Pro subscription you can use a kit : 

### 1. Kit settings

In the settings page of the kit, set the "Technology" flag to "SVG". With this setting, the JS code will load and inject the SVG code on the fly.

### 2. Load script in Nextjs

```jsx
// _app.jsx

import Script from 'next/script'

export default function MyApp({ Component, pageProps }) {
  return (
    <>
      <Script
        type="text/javascript"
        src="https://kit.fontawesome.com/[YOUR_KIT_TOKEN].js"
        crossOrigin="anonymous"
        data-auto-replace-svg="nest" // nest SVG instead of replacing the original tag
        async
      />
      <Component {...pageProps} />
    </>
  )
}
```

Note that I had to make sure the SVG is added inside the original tag instead of replacing it by setting the `data-auto-replace-svg` attribute to `nest` on the Script tag, otherwise I would get an error (`NotFoundError: Failed to execute 'removeChild' on 'Node': The node to be removed is not a child of this node`).

Also, make sure to use the `next/script` component and not the native `script` tag, otherwise you will get errors due to inconsistencies between server and client DOM.

### 3. Workaround for broken icon height

There is a known issue (https://github.com/FortAwesome/Font-Awesome/issues/17553) when using client-side routing, after transitioning the height of the icon is not limited to 1em, effectively breaking the page layouts.

The workaround I found (not ideal) is to add the ruleset for the default class (`.svg-inline--fa`) globally : 

```css
.svg-inline--fa {
  display: var(--fa-display, inline-block);
  height: 1em;
  overflow: visible;
  vertical-align: -0.125em;
}
```
